hadoop-hdfs-project/hadoop-hdfs/src/main/java/org/apache/hadoop/hdfs/server/datanode/DataNode.java

```diff
@@ -815,6 +815,25 @@ public IOException call() {
                     String.format("FAILED TO ADD: %s: %s%n",
                         volume, ioe.getMessage()));
                 LOG.error("Failed to add volume: {}", volume, ioe);
+                /**
+                 * TODO: Some cases are not supported yet with
+                 *   same-disk-tiering on. For example, when replacing a
+                 *   storage directory on same mount, we have check if same
+                 *   storage type already exists on the mount. In this case
+                 *   we need to remove existing vol first then add.
+                 *   Also, we will need to adjust new capacity ratio when
+                 *   refreshVolume in the future.
+                 */
+                if (ioe.getMessage()
```

Comment:
It may be better to check if there is conflict with the same-disk-tiering feature when we first load the refreshVolume configuration. I.e., we can do some verification on changedVolumes